app.py

```diff
@@ -4,81 +4,173 @@
+                f"lower than the current spot price {str(max_current_spot_price)}")
+        print(f"Are you willing to pay {proposed_max_spot_price} US Dollars / hour for each instance? "
+              f"Note: You have requested for {asg_size} instances of type {ec2_instance_type}.")
+        try:
+            user_consent = input("y/N: ")
+            logging.debug(f"Received user input {user_consent}")
+        except EOFError as exc:
+            logging.error(f"Ran into error, ", exc)
+            sys.exit(1)
+        if not (user_consent.lower() == "y" or user_consent.lower == "yes"):
+            print("Exiting because the proposed spot price is not acceptable.")
+            sys.exit(1)
+        # The yaml file as well as ASG require the spot price to be a string
+        max_spot_price = str(proposed_max_spot_price)
+        # Update max_spot_price in config file
+        logging.info("Attempting to update config file with the chosen spot price")
+        try:
+            # We read the config file again because the current config hash is created by
+            # merging the default hash with the user config hash
+            with open('config.yaml', 'w') as f:
+                user_config = yaml.safe_load(f)
+                user_config['aws']['max_spot_price'] = max_spot_price
+                yaml.dump(user_config, f)
+            logging.debug("Updated config file")
+        except OSError as exc:
+            print("Updating config file failed, proceeding to set up ASG ", exc)
+    logging.info(f"Using spot price {max_spot_price}")
+    return max_spot_price
+
 
 def cdk_init(stack_name: str, force_spot_price: bool):
-  config = configure(stack_name)
-  tags = config.get('tags')
-  vpc_stack = VpcStack(app, 
-                      f"{stack_name}-vpc",
-                      cidr=config['cidr'],
-                      env=core.Environment(region=config['region'],
-                      )
-                    )
-  asg_stack = AsgStack(app, 
-                    f"{stack_name}-asg", 
-                    stack_name=stack_name,
-                    region=config['region'], 
-                    vpc=vpc_stack.vpc, 
-                    ec2_instance_type=config['ec2_instance_type'], 
-                    ami_id=config['ami_id'],
-                    ssh_key=config['ssh_key'], 
-                    max_spot_price=config['max_spot_price'],
-                    ssh_allow_ip_range=config['ssh_allow_ip_range'],
-                    asg_size=config['asg_size'],
-                    force_spot_price=force_spot_price,
-                    env=core.Environment(region=config['region'])
-                  )
-  
-  for tag in tags:
-    core.Tag.add(asg_stack, tag.get('name'), tag.get('value'))  
-    core.Tag.add(vpc_stack, tag.get('name'), tag.get('value'))
-
-  app.synth()
+    config = configure(stack_name)
+    if (not force_spot_price) or (feature_query_spot_prices_disabled):
```

Comment:
I see that setting `feature_query_spot_prices_disabled` to `True` will still call `process_spot_pricing` - is that expected?

Reply:
Not expected. I suck at naming features correctly :/. Fixed the `if` condition